File: Session2/121. Best Time to Buy and Sell Stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n=prices.length;
        if(n==0)
        {
            return 0;
        }
        int profit=0;
        int mprice=prices[0];
        for(int i=1;i<n;i++)
        {
           mprice=Math.min(prices[i],mprice);
           profit=Math.max(prices[i]-mprice,profit);
        }
        return profit;
    }
}

File: Session2/121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n=len(prices)
        minprice=float('inf')
        maxprofit=-float('inf')
        for p in prices:
            minprice=min(minprice,p)
            maxprofit=max(maxprofit,p-minprice)
        return maxprofit